From 6e688d2bd1e6e1545fdb25bcfeee28f78240cbfb Mon Sep 17 00:00:00 2001
From: Fiona Klute <fiona.klute@gmx.de>
Date: Wed, 14 Oct 2026 17:31:09 +0200
Subject: [PATCH] build-sys: Allow disabling manpage build

In builds for embedded systems (e.g. Buildroot) manpages are not
needed, and building them adds a dependency on pod2man. Allow passing
"--disable-docs" to configure to disable building manpages.

Signed-off-by: Fiona Klute <fiona.klute@gmx.de>
Upstream: https://github.com/stefanberger/libtpms/pull/641
---
 Makefile.am  | 5 ++++-
 configure.ac | 6 ++++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 03ed7e65..9d0ea053 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -23,10 +23,13 @@
 
 SUBDIRS   = \
 	include \
-	man \
 	src \
 	tests
 
+if ENABLE_DOCS
+SUBDIRS += man
+endif # ENABLE_DOCS
+
 ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = \
diff --git a/configure.ac b/configure.ac
index b19db106..0fc928c8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -282,6 +282,11 @@ AC_ARG_ENABLE([test-coverage],
   ]
 )
 
+AC_ARG_ENABLE([docs],
+    [AS_HELP_STRING([--disable-docs], [Disable building documentation])],,
+    [enable_docs=yes])
+AM_CONDITIONAL(ENABLE_DOCS, [test "$enable_docs" = "yes"])
+
 LT_INIT
 AC_PROG_CC
 AC_PROG_CXX
@@ -392,5 +397,6 @@ echo "Use openssl crypto for  : $use_openssl_functions_for"
 echo "Test coverage           : $enable_test_coverage"
 echo "Static build            : $enable_static"
 echo "Statically linked tests : $enable_static_tests"
+echo "Build documentation     : $enable_docs"
 echo
 echo
-- 
2.55.0

